src/Api/AdminConsole/Controllers/OrganizationsController.cs

```diff
@@ -766,4 +777,15 @@ public async Task<OrganizationSsoResponseModel> PostSso(Guid id, [FromBody] Orga
 
         return new OrganizationSsoResponseModel(organization, _globalSettings, ssoConfig);
     }
+
+    private async Task TryGrantOwnerAccessToSecretsManagerAsync(Guid organizationId, Guid userId)
+    {
+        var organizationUser = await _organizationUserRepository.GetByOrganizationAsync(organizationId, userId);
+
+        if (organizationUser is { Status: OrganizationUserStatusType.Confirmed, Type: OrganizationUserType.Owner })
```

Comment:
Each endpoint that uses this method already has its own permission check, so I don't think we need to check for Owner again here. The Jira ticket says that anyone who's managed to add SM should be given access, so I assume if they've reached this point in the code, we should just give it to them.

It also uses slightly different logic (`CurrentContext.EditSubscription` vs. `OrganizationUserType.Owner`) which risks diverging in the future.

It's true that we don't want to give providers access to SM, but they don't use these endpoints (granted that is not obvious) and in any case would return `null` from the `organizationUser` repo call.